Orange/preprocess/util.py

```diff
@@ -0,0 +1,16 @@
+import warnings
+
+from Orange.data import Table
+from Orange.util import OrangeDeprecationWarning
+
+
+class _RefuseDataInConstructor:
+    # TODO: drop this as soon as possible
+    def __new__(cls, *args, **kwargs):
+        if (args and isinstance(args[0], Table) or
+                isinstance(kwargs.get('data'), Table)):
+            warnings.warn(
+                '%s no longer accepts data in the constructor. Instead, '
+                'first make an instance, i.e. %s(), then call it with data.'
+                % (cls, cls), OrangeDeprecationWarning, stacklevel=2)
+        return super().__new__(cls)
```

Comment:
IMHO DeprecationWarnings are used to point that a feature/method/class is going away and that a developer should modify the code as it will not work with the future versions of the library. Here, a warning is used to point that the code does not work any more.

I would rather have a ValueError raised (with the same message), or keep the old behaviour working until some future version (probably mentioned in the warning).